### PR TITLE
INCIDEN-660

### PR DESCRIPTION
**Pull request by jkunle**
## Why

see https://govukverify.atlassian.net/browse/INCIDEN-660

Similar to https://github.com/govuk-one-login/ipv-spot/pull/951

### What changed

Initialisation of SSMClient

### Why did it change

To resolve ERROR InvalidSignatureException: Signature expired: 20240129T194125Z is now earlier than 20240129T194717Z (20240129T195217Z - 5 min.)
Based on SPOT resolution to a similar error

### Issue tracking

- [Inciden-660](https://govukverify.atlassian.net/browse/Inciden-660)

### CODE REVIEW

File: lambdas/get-addresses/package.json [hunk 1]
```diff
@@ -15,7 +15,9 @@
         "@aws-sdk/client-dynamodb": "3.362.0",
         "@aws-sdk/client-ssm": "^3.362.0",
         "@aws-sdk/lib-dynamodb": "3.150.0",
-        "esbuild": "0.14.14"
+        "@smithy/node-http-handler": "2.3.1",
```
Comment (andyloughran) [inline lambdas/get-addresses/package.json:18]: Is there an approval for this package to be used in the lambda?

File: lambdas/get-addresses/src/lib/param-store-client.ts [hunk 2]
```diff
@@ -1,3 +1,16 @@
 import { SSMClient } from "@aws-sdk/client-ssm";
+import { NodeHttpHandler } from "@smithy/node-http-handler";
+import { Agent } from "https";
 
-export const SsmClient = new SSMClient({ region: "eu-west-2" });
+const DEFAULT_CONNECTION_TIMEOUT = 30000;
```
Comment (andyloughran) [inline lambdas/get-addresses/src/lib/param-store-client.ts:5]: Is this ms?

File: lambdas/get-addresses/package.json (hunk 1 above)
Comment (jkunle) [reply lambdas/get-addresses/package.json:18]: I tried install this `@aws-sdk/node-http-handler` got some depreciation warning in favor of the one above AWS SDK uses this see https://github.com/aws/aws-sdk-js-v3/blob/main/UPGRADING.md

File: lambdas/get-addresses/src/lib/param-store-client.ts (hunk 2 above)
Comment (jkunle) [reply lambdas/get-addresses/src/lib/param-store-client.ts:5]: Yeah, milliseconds

File: lambdas/get-addresses/src/lib/param-store-client.ts [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+const DEFAULT_SOCKET_TIMEOUT = 30000;
+
+const requestHandler = new NodeHttpHandler({
+    httpsAgent: new Agent(),
+    connectionTimeout: DEFAULT_CONNECTION_TIMEOUT,
+    socketTimeout: DEFAULT_SOCKET_TIMEOUT,
+});
+export const SsmClient = new SSMClient({
```
Comment (CharlesIC) [inline lambdas/get-addresses/src/lib/param-store-client.ts:13]:
```suggestion

export const SsmClient = new SSMClient({
```
Add some space for readability

**Review by CharlesIC — APPROVED** [review]
The changes look good to me code-wise 👍

**Comment by jkunle** [comment]
Cannot use the proposed solution i.e. top-level await as it is still experimental in terms of library support

**Pull request closed without merging**